dali/sasha/sadbghk.cpp

```diff
@@ -0,0 +1,179 @@
+#include "platform.h"
+
+#include "jfile.hpp"
+#include "jlib.hpp"
+#include "jlog.hpp"
+#include "jiface.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+
+#include "saserver.hpp"
+#include "sautil.hpp"
+
+static constexpr unsigned defaultExpiryDays = 14;
+static constexpr unsigned defaultHouseKeepingIntervalHours = 24;
+
+#define LOGDBGHK "DEBUGPLANEHOUSEKEEPING: "
+
+// Debug Housekeeping monitor
+
+class CSashaDebugPlaneHousekeepingServer : public ISashaServer, public Thread
+{
+    std::atomic<bool> stopped{true};
+    Semaphore stopSem;
+    Mutex runMutex;
+    StringBuffer debugDir;
+    unsigned expiryDays{0};
+    unsigned interval{0};
+
+public:
+    IMPLEMENT_IINTERFACE_USING(Thread);
+
+    CSashaDebugPlaneHousekeepingServer()
+        : Thread("CSashaDebugPlaneHousekeepingServer")
+    {
+    }
+
+    virtual void start() override
+    {
+        Owned<IPropertyTree> compConfig = getComponentConfig();
+        expiryDays = compConfig->getPropInt("@expiryDays", defaultExpiryDays);
+        if (expiryDays == 0)
+            return;
+        interval = compConfig->getPropInt("@interval", defaultHouseKeepingIntervalHours);
+        if (interval == 0)
+            return;
+
+        // get debug plane dir
+        StringBuffer planeName;
+        if (!getDefaultPlane(planeName, "@debugPlane", "debug"))
+        {
+            WARNLOG("Failed to get default debug plane");
+            return;
+        }
+        Owned<IPropertyTree> plane = getStoragePlane(planeName);
+        assertex(plane);
+        verifyex(plane->getProp("@prefix", debugDir));
```

Comment:
trivial: reads odd that line 55 is an assertex and this is a verifyex. I'd change it to a single verifyex:
```
verifyex(plane && plane->getProp("@prefix", debugDir));
```